InvariantDisks/IDLogUtils.cpp

```diff
@@ -0,0 +1,162 @@
+//
+//  IDASLUtils.cpp
+//  InvariantDisks
+//
+//  Created by Gerhard Röthlin on 2015.08.01.
+//  Copyright (c) 2015 the-color-black.net. All rights reserved.
+//
+
+#include "IDLogUtils.hpp"
+
+#include <vector>
+#include <algorithm>
+
+#include <unistd.h>
+#include <fcntl.h>
+
+#ifdef ID_USE_ASL
```

Comment:
I had that in an earlier version, but in general it's much better to give the user control over such configurations via the build system. If you want better auto detection, I can add that to ID too. It'd be similar to what you wrote, but it'd define either ID_USE_ASL or ID_USE_OS_LOG, only if neither is already defined by the build system (so we'd get auto detection AND overridability)